Gulpfile.ts

```diff
@@ -0,0 +1,1125 @@
+        declaration: false,
+        removeComments: true,
+        noResolve: false,
+        stripInternal: false,
+        outFile: processDiagnosticMessagesJs
+    }, /*useBuiltCompiler*/ false);
+    return gulp.src(processDiagnosticMessagesTs)
+        .pipe(sourcemaps.init())
+        .pipe(tsc(settings))
+        .pipe(sourcemaps.write("."))
+        .pipe(gulp.dest("."));
+});
+
+// The generated diagnostics map; built for the compiler and for the "generate-diagnostics" task
+gulp.task(diagnosticInfoMapTs, [processDiagnosticMessagesJs], (done) => {
+    if (needsUpdate(diagnosticMessagesJson, [generatedDiagnosticMessagesJSON, diagnosticInfoMapTs])) {
+        exec(host, [processDiagnosticMessagesJs, diagnosticMessagesJson], done, done);
+    }
+    else {
+        done();
+    }
+});
+
+gulp.task(builtGeneratedDiagnosticMessagesJSON, [diagnosticInfoMapTs], (done) => {
+    if (fs.existsSync(builtLocalDirectory) && needsUpdate(generatedDiagnosticMessagesJSON, builtGeneratedDiagnosticMessagesJSON)) {
+        fs.writeFileSync(builtGeneratedDiagnosticMessagesJSON, fs.readFileSync(generatedDiagnosticMessagesJSON));
+    }
+    done();
+});
+
+gulp.task("generate-diagnostics", "Generates a diagnostic file in TypeScript based on an input JSON file", [diagnosticInfoMapTs]);
+
+
+const servicesFile = path.join(builtLocalDirectory, "typescriptServices.js");
+const standaloneDefinitionsFile = path.join(builtLocalDirectory, "typescriptServices.d.ts");
+const nodePackageFile = path.join(builtLocalDirectory, "typescript.js");
+const nodeDefinitionsFile = path.join(builtLocalDirectory, "typescript.d.ts");
+const nodeStandaloneDefinitionsFile = path.join(builtLocalDirectory, "typescript_standalone.d.ts");
+
+gulp.task(builtLocalCompiler, false, [servicesFile], () => {
+    const localCompilerProject = tsc.createProject("src/compiler/tsconfig.json", {typescript: require("./built/local/typescript.js")});
+    let result: NodeJS.ReadWriteStream = localCompilerProject.src()
+        .pipe(sourcemaps.init())
+        .pipe(tsc(localCompilerProject));
+    if (!useDebugMode) {
+        result = result.pipe(insert.prepend(fs.readFileSync(copyright)));
+    }
+    return result.pipe(sourcemaps.write("."))
+        .pipe(gulp.dest(builtLocalDirectory));
+});
+
+gulp.task(servicesFile, false, ["lib", "generate-diagnostics"], (done) => {
+    const settings: tsc.Settings = getCompilerSettings({
+        declaration: true,
+        preserveConstEnums: true,
+        removeComments: false,
+        noResolve: false,
+        stripInternal: true,
+        outFile: servicesFile
+    }, /*useBuiltCompiler*/ false);
+    const {js, dts} = gulp.src(servicesSources)
+        .pipe(sourcemaps.init())
+        .pipe(tsc(settings));
+    let result: NodeJS.ReadableStream = js;
+    if (!useDebugMode) {
+        result = result.pipe(insert.prepend(fs.readFileSync(copyright)));
+    }
+    result.pipe(sourcemaps.write("."))

```

Comment:
There might be a better way to structure this so that you can `return` something gulp can monitor rather than relying on `console.error`. 

Consider using something like [merge2](https://www.npmjs.com/package/merge2) to merge the events from multiple streams to report errors back to gulp.